pkg/cmdline/cmdline_linux.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package cmdline is parser for kernel command-line args from /proc/cmdline.
+//
+// It's conformant with
+// https://www.kernel.org/doc/html/v4.14/admin-guide/kernel-parameters.html,
+// though making 'var_name' and 'var-name' equivalent may need to be done
+// separately.
+package cmdline
+
+import (
+	"os"
+)
+
+const cmdLinePath = "/proc/cmdline"
+
+var procCmdLine *CmdLine
+
+func cmdLine(n string) *CmdLine {
+	r, err := os.Open(n)
+	if err != nil {
+		procCmdLine = &CmdLine{Err: err}
+		return procCmdLine
+	}
+	r.Close()
```

Comment:
should this be `defer r.Close()` ? Line 29 below calls `parse(r)` which I'd expect to fail after `r` is closed?   This tells me that maybe some test coverage is broken? Or my understanding is broken?

Reply:
Yep, I will chalk this up to only being able to test on osx while I was doing this ... thanks.